addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -488,27 +497,78 @@ def test_message_process_alias_config_bounced_to(self):
         self.assertIn(bounce_message_with_alias, self._mails[0].get('body'))
 
     @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models')
-    def test_message_process_alias_defaults(self):
+    @patch('odoo.addons.mail.models.mail_alias.Alias._on_message_alias_config_error')
+    def test_message_process_alias_defaults(self, _on_message_alias_config_error):
         """ Test alias defaults and inner values """
-        self.alias.write({
+        container_custom_value = self.env['mail.test.container'].create({})
+        self.alias_valid.write({
             'alias_user_id': self.user_employee.id,
-            'alias_defaults': "{'custom_field': 'defaults_custom'}"
+            'alias_defaults': f"{{'container_id': {container_custom_value.id}}}",
         })
 
-        record = self.format_and_process(MAIL_TEMPLATE, self.email_from, 'groups@test.com', subject='Specific')
+        self.assertEqual(self.alias_valid.alias_status, 'not_tested')
+
+        record = self.format_and_process(MAIL_TEMPLATE, self.email_from, 'valid@test.com', subject='Specific',
+                                         target_model=self.test_model_track.model)
+
+        _on_message_alias_config_error.assert_not_called()
+        self.assertEqual(self.alias_valid.alias_status, 'valid')
         self.assertEqual(len(record), 1)
         res = record.get_metadata()[0].get('create_uid') or [None]
         self.assertEqual(res[0], self.user_employee.id)
         self.assertEqual(record.name, 'Specific')
-        self.assertEqual(record.custom_field, 'defaults_custom')
+        self.assertEqual(record.container_id, container_custom_value, 'Custom value from alias_defaults')
 
-        self.alias.write({'alias_defaults': '""'})
-        record = self.format_and_process(MAIL_TEMPLATE, self.email_from, 'groups@test.com', subject='Specific2')
+        self.alias_valid.write({'alias_defaults': '""'})
+        self.assertEqual(self.alias_valid.alias_status, 'not_tested', 'Updating alias_defaults must reset status')
+        record = self.format_and_process(MAIL_TEMPLATE, self.email_from, 'valid@test.com', subject='Specific2',
+                                         target_model=self.test_model_track.model)
+
+        _on_message_alias_config_error.assert_not_called()
         self.assertEqual(len(record), 1)
         res = record.get_metadata()[0].get('create_uid') or [None]
         self.assertEqual(res[0], self.user_employee.id)
         self.assertEqual(record.name, 'Specific2')
-        self.assertFalse(record.custom_field)
+        self.assertFalse(record.container_id)
+
+        self.assertEqual(self.alias_valid.alias_status, 'valid')
+
+    @mute_logger('odoo.addons.mail.models.mail_thread', 'odoo.models', 'odoo.sql_db')
+    def test_message_process_invalid_alias_defaults(self):
```

Comment:
As we already have two test functions 'test_message_process_alias_config_bounced_content' and 'test_message_process_alias_config_bounced_to', maybe this one could be 'test_message_process_alias_config_invalid_defaults' ?

Reply:
Yes, name changed. (the idea was to name it similarly to the previous test to show that it was the failing case of the previous test (happy case) but with the changes described above, it is no longer the case)